### PR TITLE
[FIX] mail, web_editor: add attachment at file upload

### PR DESCRIPTION
**Pull request by reth-odoo**
When using the `/image` command in the composer, or otherwise uploading a file the editor should add the attachment to the composer if it is the current model

During a change in js relational models [1] the code was not adapted properly. This lead to a traceback when using the command inside the composer.

[1]: 218ad8456a06503dd508e7216edcffdc90b35cac

task-3741858

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/test_website/static/tests/field_html_file_upload.js [hunk 1]
```diff
@@ -95,13 +97,34 @@ QUnit.module('field html file upload', {
...
         };
         const mockRPC = (route, args) => {
-            if (route === "/web_editor/attachment/add_data") {
-                return Promise.resolve({"id": 5, "name": "test.jpg", "description": false, "mimetype": "image/jpeg", "checksum": "7951a43bbfb08fd742224ada280913d1897b89ab",
-                                        "url": false, "type": "binary", "res_id": 1, "res_model": "note.note", "public": false, "access_token": false,
-                                        "image_src": "/web/image/1-a0e63e61/test.jpg", "image_width": 1, "image_height": 1, "original_id": false
-                                        });
-            }
-            else if (route === "/web/dataset/call_kw/ir.attachment/generate_access_token") {
+        if (args.method === "web_save") {
```
Comment (rdeodoo) [inline addons/test_website/static/tests/field_html_file_upload.js:100]: Could you move this in `web_editor`?
This has nothing to do with `website` module, it's a fix in `web_editor` for the composer.

It also really not have anything to do with the `test_website` module.
`test_website` purpose is to be able to test behavior in test which would require real elements to be added in the DB which can't be done in a regular module, like some routes, some views, some models etc which we need to be encapsulated in a separate module and there is no workaround it.

Unless your fix has anything to do with `web_unsplash`?
Initial test here was introduced because it needed a bridge for web_editor and web_unsplash, which was okay~ish but if we introduce some fix for web_editor, let's add the test in.. web_editor please.

**Review by rdeodoo — CHANGES_REQUESTED** [review]

File: addons/test_website/static/tests/field_html_file_upload.js (hunk 1 above)
Comment (reth-odoo) [reply addons/test_website/static/tests/field_html_file_upload.js]: @rdeodoo indeed web_unsplash is why this test was added in that module to start with.

The issue is that if web_unsplash is installed at all, then I need to initialize the service so the image picker can actually open. But if it's not installed, then trying to start it will fail.

I think it's good best if tests work consistently with only the module they're in and its dependencies.

If you have some idea to conditionally import it I could go with that. Currently runbot will crash if I put the test in the "correct" file.

**Comment by tde-banana-odoo** [comment]
Ping @Goaman @dmo-odoo

**Comment by reth-odoo** [comment]
It's already from their pipe actually ^^'
(finger slipped so it pinged the wrong team, my bad)

**Comment by dmo-odoo** [comment]
LGTM in theory but, in practice, runbot is red... ^^'

**Comment by reth-odoo** [comment]
@dmo-odoo the error is linked to moving the test as I attempt to explain on the rde's comment. If you look for the cause you'll see it's that web_unsplash isn't available (because not loaded in the test, because not available in the module)

Not sure if you have an idea to somehow conditionally load a service depending on if a module is loaded, or if I should just leave the test in test_website

**Comment by dmo-odoo** [comment]
@reth-odoo I think what @rdeodoo meant is that your fix should be in a brand new test in the `web_editor` folder, not to move the entire existing test which does rely on `web_unsplash`. Technically the use case of your fix does not necessarily rely on `web_unsplash` so it should be possible to test it in `web_editor`.

**Comment by reth-odoo** [comment]
I'll keep the moving of the test and just add a patch to the test in web_unsplash so we can still open the media dialog without fuss when it's installed.
Seems sensible to me as unpatching the patches from web_unsplash isn't that easy, but let me know.

**Comment by reth-odoo** [comment]
'lil bump on this one @rdeodoo @dmo-odoo

**Comment by dmo-odoo** [comment]
I mean why not, if the entire test didn't require website, it makes sense to have it moved somewhere else. It LGTM but I'd like to have @rdeodoo give his approval as well since I still think that's not what he meant initially.

**Review by rdeodoo — APPROVED** [review]
For the test move, without looking in depth it's LGTM.
We had a discussion on discord and the initial test (introduced with https://github.com/odoo/odoo/commit/0553e16dbdc7fddd75b9f70905c32943fed6896a) which was using unsplash was actually not fixing/testing something unsplash related.
tl;dr of our discussion:
> i.e. unsplash sert juste a pas crash..

So if it's now moved in web_editor, and still works, it's nice :+1:

**Comment by dmo-odoo** [comment]
@reth-odoo Maybe change the commit title because `website_editor` is not a module (yet?)

**Comment by reth-odoo** [comment]
Thanks both :+1:
Renamed the commit ^^

> (yet?) 

:eyes: spoilers?

**Comment by dmo-odoo** [comment]
@robodoo rebase-ff r+
